README.md

```diff
@@ -1,234 +1,679 @@
+| `Label`           | Defines the structure for labels that may be predicted about a subject at specific times in the subject record.                       | PyArrow            |
+
+Below, each schema is introduced in detail. Usage examples and a practical demonstration with the [MIMIC-IV demo](https://physionet.org/content/mimic-iv-demo/2.2/) dataset are provided in a later section.
+
+> [!IMPORTANT]
+> Each component is implemented as a Schema class via the
+> [`flexible_schema`](https://flexible-schema.readthedocs.io/en/latest/) package. This allows us to capture
+> the fact that our schemas often are _open_ (they allow extra columns) and have optional columns (columns
+> whose presence or absence does not affect the validity of a dataset). This package also provides convenient
+> accessors to column names and dtypes as well as table / schema validation or alignment functionality. Under
+> the hood, all schemas are still simple standard PyArrow schemas or JSON schemas, as indicated.
+
+### The `Data` schema
+
+The `Data` schema describes a structure for the underlying medical data. It contains the following columns:
+
+| **Column Name** | **Conceptual Description**                                                                                                          | **Type**             | **Required** | **Nullable**                                           |
+| --------------- | ----------------------------------------------------------------------------------------------------------------------------------- | -------------------- | ------------ | ------------------------------------------------------ |
+| `subject_id`    | The ID of the subject this measurement is about.                                                                                    | `pa.int64()`         | Yes          | No                                                     |
+| `time`          | The time of the measurement.                                                                                                        | `pa.timestamp('us')` | Yes          | Yes, for static measurements                           |
+| `code`          | The primary categorical descriptor of what is being measured. E.g., the laboratory test being measured or diagnosis being recorded. | `pa.string()`        | Yes          | No                                                     |
+| `numeric_value` | Any numeric value associated with this measurement (e.g., the laboratory test result).                                              | `pa.float32()`       | No           | Yes, for measurements that do no have a numeric value. |
+| `text_value`    | Any text value associated with this measurement (e.g., the result of a text-based test, a clinical note).                           | `pa.large_string()`  | No           | Yes, for measurements that do not have a text value.   |
+
+In addition, the `Data` schema is _open_, meaning it can contain any number of custom columns to further
```

Comment:
Is there any canonical guidance/recommendation for representing categorical and boolean values in the MEDS dataset? 

As I understand the "proper" way would be to add that them as codes with the format `CODE_TYPE//CATEGORICAL_VALUE` and `CODE_TYPE//BOOLEAN_VALUE` (?) but it could also be possible force this into numeric or text values (but then there would be no way to represent that some code only has a limited range of possible values). Or is this completely up to the dataset designer, with no standard in that sense?

Reply:
No current standardization, but that may change in the future pending community usage. but for now it is intentionally un-standardized.